### PR TITLE
Persist 'owned' addresses in IPAM

### PR DESCRIPTION
**Pull request by bboreham**
New take on #1945, following on from #1971 

Here we are serializing the entire set every time it changes, which is overkill, but we don't expect the set to have more than 1,000 entries on one host and we don't expect it to change more than ~10 times per second.

[Review on Reviewable](https://reviewable.io/reviews/weaveworks/weave/1973)

### CODE REVIEW

**Comment by bboreham** [comment]
I removed the test for addresses added with `weave attach`; it requires serious modification to proxy startup to be able to attach containers started outside of the proxy.  I think we can ship 1.5 without that feature.

File: test/163_docker_restart_test.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        sleep 1
+    done
+    echo "Timed out waiting for proxy to start" >&2
+    exit 1
+}
+
+start_suite "Containers get same IP address on restart"
+
+# Remove any persisted data from previous runs
+run_on $HOST1 "sudo rm -f /tmp/test645-*"
```
Comment (awh) [inline test/163_docker_restart_test.sh]: Should be 163

File: test/163_docker_restart_test.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+WEAVE_DOCKER_ARGS="-v /tmp:/db --restart=always" weave_on $HOST1 launch-router --db-prefix=/db/test645-
```
Comment (awh) [inline test/163_docker_restart_test.sh]: Should be 163

**Comment by bboreham** [comment]
I have rebased, fixed the test directory name, and moved half the commits out to #2012

**Pull request merged**